### PR TITLE
udevd: replace udev_device by sd_device

### PR DESCRIPTION
**Pull request by yuwata**
~~This is blocked by #10603. Also, this needs to be reworked if #10722 is accepted.~~

### CODE REVIEW

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging b2496e01740a50c632b83fe486f153910332b098 into 911efe0fbf23ac51b2e8516dc02df3f8973ff758 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-d297a6632394d1659f942610ca7d2937771834d2)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging b2496e01740a50c632b83fe486f153910332b098 into 33dbab6fde5423ce61b5544ca49afa13495477a8 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-19c535d84d3d26323a3550319f74286965d5cf3c)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging 431fab3d10b90b986df34267f1f4d62a6328479e into 33dbab6fde5423ce61b5544ca49afa13495477a8 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-dd5c6fbe0f740003a091d7a2558423eb8390afda)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by yuwata** [comment]
#10603 and #10722 are merged. This is rebased on them. PTAL.

File: src/udev/udevd.c [hunk 1]
```diff
@@ -343,30 +343,52 @@ static int worker_lock_block_device(struct udev_device *dev, int *ret_fd) {
...
-        if (startswith(sysname, "dm-") ||
-            startswith(sysname, "md") ||
-            startswith(sysname, "drbd"))
+        r = sd_device_get_sysname(dev, &val);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to get sysname: %m");
+
+        if (startswith(val, "dm-") ||
+            startswith(val, "md") ||
+            startswith(val, "drbd"))
```
Comment (poettering) [inline src/udev/udevd.c:363]: (not directly on topic, but I think we should probably add STARTSWITH_SET() to make this prettier, similar to how we already have PATH_STARTSWITH_SET(), STR_IN_SET()…)

**Pull request merged**